scripts/generate_metric_enums.py

```diff
@@ -0,0 +1,144 @@
+# Script that takes src/include/duckdb/common/enums/optimizer_type.hpp, extracts the optimizer types
+# and adds them to the metrics types.
+# Then it creates a new file src/include/duckdb/common/enums/metric_type.hpp with the new metrics types as enums.
+# "Generate_enums_util.py" is then run to update the enums.hpp file
+
+import re
+import os
+
+os.chdir(os.path.dirname(__file__))
+
+metrics_header_file = os.path.join("..", "src", "include", "duckdb", "common", "enums", "metric_type.hpp")
+metrics_cpp_file = os.path.join("..", "src", "common", "enums", "metric_type.cpp")
+optimizer_file = os.path.join("..", "src", "include", "duckdb", "common", "enums", "optimizer_type.hpp")
+
+metrics = [
+    "QUERY_NAME",
+    "IDLE_THREAD_TIME",
+    "CPU_TIME",
+    "EXTRA_INFO",
+    "CUMULATIVE_CARDINALITY",
+    "OPERATOR_TYPE",
+    "OPERATOR_CARDINALITY",
+    "CUMULATIVE_ROWS_SCANNED",
+    "OPERATOR_ROWS_SCANNED",
+    "OPERATOR_TIMING",
+    "ALL_OPTIMIZERS",
+    "CUMULATIVE_OPTIMIZER_TIMING",
+    "PLANNER_TIMING",
+    "PLANNER_BINDING_TIMING",
+    "PHYSICAL_PLANNER_TIMING",
+    "PHYSICAL_PLANNER_COLUMN_BINDING_TIMING",
+    "PHYSICAL_PLANNER_RESOLVE_TYPES_TIMING",
+    "PHYSICAL_PLANNER_CREATE_PLAN_TIMING",
+]
+
+optimizer_types = []
+
+# Regular expression to match the enum values
+enum_pattern = r'\s*([A-Z_]+)\s*=\s*\d+,?|\s*([A-Z_]+),?'
+
+inside_enum = False
+
+# open the optimizer file and extract the optimizer types
+with open(optimizer_file, "r") as f:
+    for line in f:
+        line = line.strip()
+
+        if line.startswith("enum class OptimizerType"):
+            inside_enum = True
+            continue
+
+        if inside_enum and line.startswith("};"):
+            break
+
+        if inside_enum:
+            match = re.match(enum_pattern, line)
+            if match:
+                optimizer_type = match[1] if match[1] else match[2]
+                if optimizer_type == "INVALID":
+                    continue
+                optimizer_types.append(optimizer_type)
+
+header = """//-------------------------------------------------------------------------
+//                         DuckDB
+//
+//
+// duckdb/common/enums/metrics_type.hpp
+// 
+// This file is automatically generated by scripts/generate_metric_enums.py
+// Do not edit this file manually, your changes will be overwritten
+//-------------------------------------------------------------------------\n
+"""
+
+typedefs = """struct MetricsTypeHashFunction {
+	uint64_t operator()(const MetricsType &index) const {
+		return std::hash<uint8_t>()(static_cast<uint8_t>(index));
+	}
+};
+
+typedef unordered_set<MetricsType, MetricsTypeHashFunction> profiler_settings_t;
+typedef unordered_map<MetricsType, Value, MetricsTypeHashFunction> profiler_metrics_t;
```

Comment:
Should we keep this in the file or move it to another file? It is not generated, strictly speaking. We might also be able to remove one of the below includes.

Reply:
Because `metric_types.hpp` is included in `profiling_info.hpp` it was causing compilation errors with the functions generated by this file.